packages/modules/influxdb/src/influxdb-container.ts

```diff
@@ -0,0 +1,242 @@
+      DOCKER_INFLUXDB_INIT_USERNAME: this.username,
+      DOCKER_INFLUXDB_INIT_PASSWORD: this.password,
+      DOCKER_INFLUXDB_INIT_ORG: this.organization,
+      DOCKER_INFLUXDB_INIT_BUCKET: this.bucket,
+    };
+
+    if (this.retention) {
+      env.DOCKER_INFLUXDB_INIT_RETENTION = this.retention;
+    }
+
+    if (this.adminToken) {
+      env.DOCKER_INFLUXDB_INIT_ADMIN_TOKEN = this.adminToken;
+    }
+
+    this.withEnvironment(env);
+  }
+
+  private configureInfluxDB1(): void {
+    const env: Record<string, string> = {
+      INFLUXDB_USER: this.username,
+      INFLUXDB_USER_PASSWORD: this.password,
+      INFLUXDB_HTTP_AUTH_ENABLED: String(this.authEnabled),
+      INFLUXDB_ADMIN_USER: this.admin,
+      INFLUXDB_ADMIN_PASSWORD: this.adminPassword,
+    };
+
+    if (this.database) {
+      env.INFLUXDB_DB = this.database;
+    }
+
+    this.withEnvironment(env);
+  }
+
+  private isInfluxDB2(tag: string): boolean {
+    if (tag === "latest") {
+      return true; // Assume latest is v2
+    }
+
+    // Parse version number
+    const versionMatch = tag.match(/^(\d+)(?:\.(\d+))?/);
+    if (versionMatch) {
+      const majorVersion = parseInt(versionMatch[1], 10);
+      return majorVersion >= 2;
+    }
+
+    return true; // Default to v2 if unable to parse
+  }
+}
+
+export class StartedInfluxDBContainer extends AbstractStartedContainer {
+  constructor(
+    startedTestContainer: StartedTestContainer,
+    private readonly username: string,
+    private readonly password: string,
+    private readonly organization: string,
+    private readonly bucket: string,
+    private readonly database?: string,
+    private readonly adminToken?: string,
+    private readonly isVersion2: boolean = true
+  ) {
+    super(startedTestContainer);
+  }
+
+  public getPort(): number {
+    return this.getMappedPort(INFLUXDB_PORT);
+  }
+
+  public getUsername(): string {
+    return this.username;
+  }
+
+  public getPassword(): string {
+    return this.password;
+  }
+
+  public getOrganization(): string {
+    return this.organization;
+  }
+
+  public getBucket(): string {
+    return this.bucket;
+  }
+
+  public getDatabase(): string | undefined {
+    return this.database;
+  }
+
+  public getAdminToken(): string | undefined {
+    return this.adminToken;
+  }
+
+  public isInfluxDB2(): boolean {
+    return this.isVersion2;
+  }
+
+  /**
+   * @returns The URL to connect to InfluxDB
+   */
+  public getUrl(): string {
+    return `http://${this.getHost()}:${this.getPort()}`;
+  }
+
+  /**
+   * @returns A connection string for InfluxDB
+   */
+  public getConnectionString(): string {
+    if (this.isVersion2) {
+      const params = new URLSearchParams({
+        org: this.organization,
+        bucket: this.bucket,
+      });
+
+      if (this.adminToken) {
+        params.append("token", this.adminToken);
+      }
+
+      return `${this.getUrl()}?${params.toString()}`;
+    } else {
+      // InfluxDB 1.x connection string format
```

Comment:
I wonder whether it would be better to have an `InfluxDBContainer` interface, and have v1 and vX implementations instead of almost every method having an if/else. Perhaps for the future